glean-core/src/metrics/experiment.rs

```diff
@@ -101,23 +117,31 @@ impl ExperimentMetric {
         }
 
         // Make sure that branch id is within the expected limit.
-        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_IDS_LEN {
+        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_BRANCH_LEN {
             log::warn!(
                 "Value length {} for branch exceeds maximum of {}",
                 branch.len(),
-                MAX_EXPERIMENTS_IDS_LEN
+                MAX_EXPERIMENTS_BRANCH_LEN
             );
-            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_IDS_LEN)
+            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_BRANCH_LEN)
         } else {
             branch
         };
 
-        // TODO (bug 1566001): add sane limits for the extra keys and
-        // values as well.
+        // Apply limits to extras
+        let truncated_extras = extra.and_then(|extra| {
+            let mut temp_map = HashMap::new();
+            for (key, value) in extra.into_iter().take(MAX_EXPERIMENTS_EXTRAS_LEN) {
+                let truncated_key = truncate_string_at_boundary(key, MAX_EXPERIMENT_KEY_LEN);
```

Comment:
In this PR we should, at the very least, show a warning in logcat and not silently truncate. Moreover, I agree with mike: let's file a follow-up bug for reporting this with the Glean SDK error reporting mechanism: this will allow, in the future, to make debug builds break if limits are hit.

Once the bug is filed, let's leave a comment here with the bug number.

Reply:
Good idea.  Bug filed [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1574468)